packages/dev/s2-docs/pages/react-aria/customization.mdx

```diff
@@ -0,0 +1,352 @@
+}
+
+<Stepper>
+  <Button slot="increment">⬆</Button>
+  <Button slot="decrement">⬇</Button>
+</Stepper>
+```
+
+#### Default slot
+
+The default slot is used to provide props to a component without specifying a slot name. This is used by children without a `slot` prop. This example passes a specific class name to a standard button child and to a button child with a slot named "end".
+
+```tsx
+import {Button, ButtonContext, DEFAULT_SLOT} from 'react-aria-components';
+
+function MyCustomComponent({children}) {
+  return (
+    <ButtonContext.Provider
+      value={{
+        slots: {
+          [DEFAULT_SLOT]: {
+            className: "default-button"
+          },
+          end: {
+            className: "end-button"
+          }
+        }
+      }}>
+      {children}
+    </ButtonContext.Provider>
+  );
+}
+
+<MyCustomComponent>
+  {/* Consumes the props passed to the default slot */}
+  <Button>Click me</Button>
+  {/* Consumes the props passed to the "end" slot */}
+  <Button slot="end">Click me</Button>
+</MyCustomComponent>
+```
+
+### Provider
+
+The `Provider` component is a utility that makes it easier to provide multiple React contexts without manually nesting them. This can be achieved by passing pairs of contexts and values as an array to the `values` prop.
+
+```tsx
+import {Provider, ButtonContext, InputContext} from 'react-aria-components';
+
+<Provider
+  values={[
+    [ButtonContext, {/* ... */}],
+    [InputContext, {/* ... */}]
+  ]}>
+  {/* ... */}
+</Provider>
+```
+
+This is equivalent to:
+
+```tsx
+<ButtonContext.Provider value={{/* ... */}}>
+  <InputContext.Provider value={{/* ... */}}>
+    {/* ... */}
+  </InputContext.Provider>
+</ButtonContext.Provider>
+```
+
+### Consuming contexts
+
+You can also consume from contexts provided by React Aria Components in your own custom components. This allows you to replace a component used as part of a larger pattern with a custom implementation. For example, you could consume from `LabelContext` in a custom label component to make it compatible with React Aria Components.
+
+#### useContextProps
+
+The <TypeLink links={docs.links} type={docs.exports.useContextProps} /> hook merges the local props with the ones provided via context by a parent component. The local props always take precedence over the context values (see [mergeProps](mergeProps.html)). `useContextProps` supports the [slot](#slots) prop to indicate which value to consume from context.
+
+```tsx
+import type {LabelProps} from 'react-aria-components';
+import {LabelContext, useContextProps} from 'react-aria-components';
+
+const MyCustomLabel = React.forwardRef(
+  (props: LabelProps, ref: React.ForwardedRef<HTMLLabelElement>) => {
+    // Merge the local props and ref with the ones provided via context.
+    /*- begin highlight -*/
+    [props, ref] = useContextProps(props, ref, LabelContext);
+    /*- end highlight -*/
+
+    // ... your existing Label component
+    return <label {...props} ref={ref} />;
```

Comment:
```suggestion
    let [allProps, mergedRef] = useContextProps(props, ref, LabelContext);
    /*- end highlight -*/

    // ... your existing Label component
    return <label {...allProps} ref={mergedRef} />;
```

this was a violation of React's new lint rules for the compiler, we should change it